### PR TITLE
wallet: add fee rate estimation to FundRawTx

### PR DESCRIPTION
**Pull request by YusukeShimizu**
This is a fix for the inconvenience caused by the `FundRawTx` , which sometimes set relatively high fees.

Add `getFeeRate` function to estimate optimal fee rate based on network conditions.
Replace FundRawTx with FundRawWithOptions to support fee rate options.

### CODE REVIEW

**Comment by grubles** [comment]
@nepet Can you take a look?

**Comment by YusukeShimizu** [comment]
In order to address this issue, I reverted to draft status

https://discord.com/channels/905126649224388629/905126649698320406/1265218998241591412

**Review by nepet — APPROVED** [review]
I think this PR is fine and I would merge it as is and start a new one to figure out the `1000 sat` problem discussed in discord https://discord.com/channels/905126649224388629/905126649698320406/1265218998241591412.
*whatever you prefere is fine tho ^^ !

ack 6904f99ab94ccf472c7d2e7473cc2beae14743ce

**Comment by YusukeShimizu** [comment]
Thank you for your review.    
For ease of test, I addressed [this](https://discord.com/channels/905126649224388629/905126649698320406/1265218998241591412) in this PR.

The pre-conditions in swap in and out have been updated to ⁠wallet_balance_sat > swap_amount_sat + estimated_fee.  
Additionally, I have integrated this check into the swap package layer as part of a refactor.  
https://github.com/ElementsProject/peerswap/pull/302/commits/a973fab764eed84e639f35a25c33c4ee8856560b

I have tested locally.

```sh
./out/pscli lbtc-getbalance
{
  "sat_amount":  "1000000"
}
```

```sh
./out/pscli swapin --channel_id 120946279120897 --sat_amt 1000000 --asset lbtc
2024/07/28 13:51:50 rpc error: code = Unknown desc = exceeding maximum swap amount: 999700
```

```sh
./out/pscli swapin --channel_id 120946279120897 --sat_amt 999700 --asset lbtc
{
  "swap":  {
    "id":  "dd926863b50200f0e5ccf4311ebbb5b0ea20e96c9a709ea2e45c72919b891ed8",
    "created_at":  "1722142344",
    "asset":  "lbtc",
    "type":  "swap-in",
    "role":  "sender",
    "state":  "State_SwapInSender_SendTxBroadcastedMessage",
    "initiator_node_id":  "0328bf2621a6ed6b4c6039c6ffe421bf8b1700b5e8136b717f9fa7953bc06eaafb",
    "peer_node_id":  "02103606f048e9680b03e70e3e7bea6ca3e4619dcfd9104112e95835845fdf556d",
    "amount":  "999700",
    "channel_id":  "110:1:1",
    "opening_tx_id":  " ",
    "claim_tx_id":  "",
    "cancel_message":  "",
    "lnd_chan_id":  "120946279120897"
  }
}
```

```sh
./bin/elements-cli -rpcwallet=peerswap1 getrawtransaction b6d03a002e2907efc5df7dc6f5e1c61ffac29ef6a03ea4d49c9ddd32011a6c09 true | jq -r '.vsize, .fee["5ac9f65c0efcc4775e0baec4ec03abdde22473cd3cf33c0419ca290e0751b225"]' | {
  read -r vsize;
  read -r fee;
  fee_satoshi=$(echo "$fee * 100000000" | bc);
  feerate=$(echo "scale=2; $fee_satoshi / $vsize" | bc);
  echo "Fee Rate: $feerate sat/vB";
  echo "Fee: $fee_satoshi sat";
}
Fee Rate: .22 sat/vB
Fee: 300.00000000 sat
```

File: wallet/elementsrpcwallet.go [hunk 1]
```diff
@@ -104,6 +108,23 @@ func (r *ElementsRpcWallet) CreateAndBroadcastTransaction(swapParams *swap.Openi
...
+	// This value is equivalent to 0.1 sat/byte.
+	minFeeRateBTCPerKb = 0.000001
+)
+
+// getFeeRate retrieves the optimal fee rate based on the current Liquid network conditions.
+// Returns the recommended fee rate in BTC/kB
+func (r *ElementsRpcWallet) getFeeRate() float64 {
+	feeRes, err := r.rpcClient.EstimateFee(LiquidTargetBlocks, "ECONOMICAL")
+	if err != nil || len(feeRes.Errors) > 0 {
+		// Return the minimum fee rate in case of an error
```
[the remark this answers is not in the document]
Comment (YusukeShimizu) [reply wallet/elementsrpcwallet.go:128]: Added https://github.com/ElementsProject/peerswap/pull/302/commits/d1e54b56eeb75fa40c7afb77bd0062ca76e53af9

**Review by nepet — APPROVED** [review]
lgtm! I appreciate that you also clean-up and refactor old stuff on the way.
Please squash the fixup commit before before merging (Let me know if you want me to squash and merge).

**Pull request merged**